filament/backend/include/private/backend/CommandStream.h

```diff
@@ -145,8 +145,20 @@ struct CommandType<void (Driver::*)(ARGS...)> {
         using SavedParameters = std::tuple<typename std::decay<ARGS>::type...>;
         SavedParameters mArgs;
 
-        void log() noexcept;
-        template<std::size_t... I> void log(std::index_sequence<I...>) noexcept;
+        // these two log() template functions should be defined inline in this class 
+        // definition, otherwise when building with msvc they are undefined. Alternatively they 
+        // should be defined in a file which calls them, not CommandStream.cpp.
+        void log() noexcept { log(std::make_index_sequence<std::tuple_size<SavedParameters>::value>{}); }
+
+        template<std::size_t... I> void log(std::index_sequence<I...>) noexcept {
+#if DEBUG_COMMAND_STREAM
+            static_assert(UTILS_HAS_RTTI, "DEBUG_COMMAND_STREAM can only be used with RTTI");
+            std::string command = utils::CallStack::demangleTypeName(typeid(Command).name()).c_str();
```

Comment:
uhm I think there is a problem here. Now the header file depends on <string>, and we don't want headers to include <string>. I'm not sure why having the definition in the .cpp doesn't work, since that's the only place it's used from?

Oh... I think maybe all you need is to surround them with `#if DEBUG_COMMAND_STREAM` in the cpp (or hide the declaration when DEBUG_COMMAND_STREAM is not used.

Reply:
@pixelflinger I'll try this

Reply:
@pixelflinger thanks for pointing me to the right direction there was a call to log() protected by a if statement instead of an #ifdef. I have updated the PR.

Reply:
@pixelflinger are you OK with the change?